src/components/radio-group/radio-button.tsx

```diff
@@ -1,115 +1,134 @@
+                {...rootProps}
+                onClick={this.onRootClick}
+                cssStates={cssStates}
+                role="radio"
+                aria-checked={this.props.checked}
             >
-                {
-                    this.props.location === 'left' && (
-                        <span className={style.leftLabel} data-automation-id="LABEL">{this.props.value}</span>
-                    )
-                }
-                <div
-                    data-automation-id="INPUT_CONTAINER"
-                    className={this.props.disabled ? style.disabled : style.enabled}
-                >
-                    {
-                        this.props.checked ?
-                            checkedRadioSvg(this.props.disabled!) :
-                            emptyRadioSvg(this.props.disabled!)
-                    }
-                    <input
-                        type="radio"
-                        className={style['radio-input']}
-                        data-automation-id={'INPUT'}
-                        value={this.props.value}
-                        checked={this.props.checked}
-                        name={this.props.name}
-                        disabled={this.props.disabled}
-                        readOnly={true}
-                    />
+                <input
+                    {...restOfProps}
+                    type="radio"
+                    className="radioInput"
+                    data-automation-id="NATIVE_INPUT"
+                    ref={ref => this.inputRef = ref}
+                    onChange={this.onInputChange}
+                    onFocus={this.onInputFocus}
+                    onBlur={this.onInputBlur}
+                />
+                <div className="contentContainer">
+                    <div
+                        data-automation-id="INPUT_CONTAINER"
+                        className="iconContainer"
+                    >
+                        {this.props.checked ? checkedRadioSvg() : emptyRadioSvg()}
+                    </div>
+                    <span className="radioLabel" data-automation-id="LABEL">{this.props.value}</span>
+                    {this.props.children}
                 </div>
-                {
-                    this.props.location === 'right' && (
-                        <span
-                            className={style.rightLabel}
-                            data-automation-id="LABEL"
-                        >
-                            {this.props.value}
-                        </span>
-                    )
-                }
-                {this.props.children}
             </div>
         );
     }
 
-    @action
-    private onClick: React.EventHandler<React.SyntheticEvent<HTMLDivElement>> = () => {
-        if (!this.props.disabled) {
-            this.props.onClick!(this.props.value);
+    private onRootClick = (e: React.MouseEvent<HTMLDivElement>) => {
+        this.inputRef && this.inputRef.click();
```

Comment:
IMHO, this is a problematic pattern. We should trickle the state of the radio button from above (controlled component). Let's discuss it tomorrow.